web/modules/custom/collation_fixer/src/CollationFixer.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\collation_fixer;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Database\StatementInterface;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\Sql\SqlEntityStorageInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+
+/**
+ * Core class for fixing collations.
+ */
+final class CollationFixer {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    private readonly Connection $connection,
+    private readonly ModuleHandlerInterface $moduleHandler,
+    private readonly EntityTypeManagerInterface $entityTypeManager,
+    private readonly EntityFieldManagerInterface $fieldManager,
+  ) {}
+
+  /**
+   * Check table collations.
+   *
+   * @param string $table
+   *   Name of a table to check for collation correctness.
+   *   Defaults to all tables if not set.
+   *
+   * @return TableCollation[]
+   *   A list of tables with wrong collation or charset.
+   */
+  public function checkCollation(?string $table = NULL): array {
+    $schema = $this->getSchema($table);
+
+    $database_name = &drupal_static(__FUNCTION__);
```

Comment:
Du skifter mellem at bruge snake_case og camelCase til variabelnavne i klassen her. Jeg tænker det er pænere at være konsistent (og jeg ville vælge camelCase i PHP-klasser, men det er mindre vigtigt end at være konsistent).

Reply:
Har Drupal cs ikke en holdning til variable navne?
Synes altid det er forvirrende fordi properties på klasserne er camel case, ikke?
Som jeg husker Drupals standard er det:
Metoder/properties: camel case
Variabler i metoder: snake case

Reply:
[Drupal Coding Standards](https://www.drupal.org/docs/develop/standards/php/php-coding-standards#s-functions-and-variables) say:

> Be consistent; do not mix camelCase and snake_case variable naming inside a file.

I have updated this to use camelCase.